pkg/resource/kubernetes.go

```diff
@@ -316,11 +162,11 @@ func ForControllerClient(client controllerClient.Client, namespace string, objTy
 	}
 }
 
-func ForListableControllerClient(client controllerClient.Client, namespace string, objType controllerClient.Object,
+func ForListableControllerClient[T controllerClient.Object](client controllerClient.Client, namespace string, objType T,
```

Comment:
We can drop `objType` and `listType` here.

Reply:
We can't - we need the actual instances.